crates/executor/src/blockchain_tree/mod.rs

```diff
@@ -212,26 +214,33 @@ impl<DB: Database, C: Consensus, EF: ExecutorFactory> BlockchainTree<DB, C, EF>
         &mut self,
         block: SealedBlockWithSenders,
     ) -> Result<BlockStatus, Error> {
-        let canonical_block_hashes = self.block_indices.canonical_chain();
-        let (_, canonical_tip) =
-            canonical_block_hashes.last_key_value().map(|(i, j)| (*i, *j)).unwrap_or_default();
-
-        // create state provider
         let db = self.externals.shareable_db();
-        let parent_header = db
-            .header(&block.parent_hash)?
+
+        // Validate that the block is post merge
+        let parent_td = db
+            .header_td(&block.parent_hash)?
             .ok_or(ExecError::CanonicalChain { block_hash: block.parent_hash })?;
+        if !self.externals.chain_spec.fork(Hardfork::Paris).active_at_ttd(parent_td, U256::ZERO) {
```

Comment:
it's functioning as intended, will add a comment to clarify. the reason `active_at_ttd` accepts td and difficulty is so it can compute the parent td. here we are just bypassing that by passing in the parent td and `0`